### PR TITLE
unify stream output approaches

### PR DESCRIPTION
**Pull request by anatolysergeev**

### CODE REVIEW

File: redis/src/main/scala/zio/redis/options/Streams.scala [hunk 1]
```diff
@@ -59,6 +59,8 @@ trait Streams {
 
   sealed case class StreamEntry[I, K, V](id: I, fields: Map[K, V])
 
+  sealed case class Stream[N, I, K, V](name: N, entries: Chunk[StreamEntry[I, K, V]])
```
Comment (anatolysergeev) [inline redis/src/main/scala/zio/redis/options/Streams.scala:62]: This class name is arguable. I accept proposals for better))
It has conflict wit zio.stream.Stream, I have to fix imports((

Comment (anatolysergeev) [reply redis/src/main/scala/zio/redis/options/Streams.scala]: StreamChunk now

File: redis/src/main/scala/zio/redis/Output.scala [hunk 2]
```diff
@@ -318,29 +318,41 @@ object Output {
...
-            case RespValue.Array(Seq(key @ RespValue.BulkString(_), value)) =>
-              outK.unsafeDecode(key) -> outV.unsafeDecode(value)
-            case other =>
-              throw ProtocolError(s"$other isn't a valid array")
-          }.toMap
-        case other => throw ProtocolError(s"$other isn't an array")
+        case RespValue.Array(Seq(id @ RespValue.BulkString(_), value)) =>
+          val eId   = ArbitraryOutput[I]().unsafeDecode(id)
+          val entry = KeyValueOutput(ArbitraryOutput[K](), ArbitraryOutput[V]()).unsafeDecode(value)
+          StreamEntry(id = eId, entry)
```
Comment (mijicd) [inline redis/src/main/scala/zio/redis/Output.scala:331]:
```suggestion
          StreamEntry(eId, entry)
```
I'd also rename it to `entryId`.

**Review by mijicd — COMMENTED** [review]
Super small remark from my side, other than that good to go!

**Review by mijicd — APPROVED** [review]

**Pull request merged**